crates/ruff_python_parser/src/lexer.rs

```diff
@@ -551,17 +551,20 @@ impl<'source> Lexer<'source> {
                     })
                 }
                 '\\' => {
-                    self.cursor.bump();
+                    self.cursor.bump(); // '\'
                     if matches!(self.cursor.first(), '{' | '}') {
                         // Don't consume `{` or `}` as we want them to be consumed as tokens.
                         break;
                     } else if !context.is_raw_string {
                         if self.cursor.first() == 'N' && self.cursor.second() == '{' {
-                            self.cursor.bump();
-                            self.cursor.bump();
+                            self.cursor.bump(); // 'N'
+                            self.cursor.bump(); // '{'
                             in_named_unicode = true;
```

Comment:
Do we want to add `eat_if` and `eat_if2` methods?

Reply:
Actually, we could use `eat_char` in the same order as it will short circuit if the first char isn't `N`.